fn_grpc_interface/fn_grpc_interface/components/function_grpc.py

```diff
@@ -0,0 +1,314 @@
+# (c) Copyright IBM Corp. 2010, 2019. All Rights Reserved.
+# -*- coding: utf-8 -*-
+# pragma pylint: disable=unused-argument, no-self-use
+"""Function implementation"""
+
+import logging
+from resilient_circuits import ResilientComponent, function, handler, StatusMessage, FunctionResult, FunctionError
+import sys
+import os
+import json
+import grpc
+import re
+import inspect
+from google.protobuf.json_format import MessageToDict
+import six
+
+try:
+    # PY3
+    import importlib.util
+except ImportError as e:
+    # PY2
+    logging.debug("The Import Error raised for the module importlib trying with imp module : {}".format(e))
+    import imp
+
+if six.PY2:
+    # Used to Convert unicode strings to utf-8 for python 2.7
+    import ast
+
+log = logging.getLogger(__name__)
+
+
+class FunctionComponent(ResilientComponent):
+    """Component that implements generic wrapper for the gRPC client on the Resilient platform"""
+
+    def __init__(self, opts):
+        """constructor provides access to the configuration options"""
+        super(FunctionComponent, self).__init__(opts)
+        self.options = opts.get("fn_grpc_interface", {})
+
+    @handler("reload")
+    def _reload(self, event, opts):
+        """Configuration options have changed, save new values"""
+        self.options = opts.get("fn_grpc_interface", {})
+
+    def _get_grpc_interface_module(self, interface_files=[], base_dir=None, files_dir=None):
```

Comment:
* These functions that have self as the first parameter should be defined inside the FunctionComponent similar to `_get_method_from_stub_object`.
* We discussed this